### DISCUSSION TITLE
Add `buildInputs` but for `cargo-package-deps` not the toplevel derivation.

### DISCUSSION THREAD

**Discussion by zmrocze** [ipetkov/crane#642] [Q&A] [answered]
I know my crate doesn't build because of system dependencies in the rust dependency for my project.

It does build if I install those system dependencies, that is:

```bash
nix-shell -p alsa-lib pkg-config -c "cargo build"
```

succeeds, but `cargo build` itself doesn't.

So now I'd like to build with crane. But adding `buildInputs = [alsa-lib pkg-config]` to the arguments to `buildDepsOnly` doesn't  do the trick. I think because it only overrides deps in the toplevel built derivation.

**Comment by zmrocze** [comment]
i think i just need to specify it as `nativeBuildInputs`?

**Comment by ipetkov** [comment, marked as answer]
Hi @zmrocze ! Usually `pkg-config` needs to go in `nativeBuildInputs` and everything else usually goes in `buildInputs` in case that helps get your build to work.

As for doing a "vanilla" `cargo build` you will likely need to enter a devshell (e.g. via `nix develop`) to pull those dependencies in. Take a look at the [sqlx example here](https://github.com/ipetkov/crane/blob/17d9e9dedd58dde2c562a4296934c6d6a0844534/examples/sqlx/flake.nix) as a quick starting point on how to set up `nativeBuildInputs`/`buildInputs` and a devshell. Hope this helps!
